src/imgui/Shortcuts.cc

```diff
@@ -0,0 +1,154 @@
+#include "Shortcuts.hh"
+#include "imgui_internal.h"
+
+#include "SettingsConfig.hh"
+
+#include <array>
+
+namespace openmsx {
+
+//-- (Only) the table in this section will be edited when adding new shortcuts
+
+struct AllShortcutInfo {
+    ShortcutIndex index;
+    ImGuiKeyChord keyChord;
+    bool local;
+    bool repeat;
+    std::string_view name; // used in settings.xml
+    std::string_view description; // shown in GUI
+};
+
+// When adding a new Shortcut, we only:
+//  * Add a new value in the above 'enum ShortcutIndex'
+//  * Add a single line in this table
+// Note: if you look at the generated code on the right, this table does NOT end up in the generated code
+//       instead it's only used to calculate various other tables (that do end up in the generated code)
+//       those 'calculations' are all done at compile-time
+// Note: the first column is ONLY used to verify (at-compile-time) whether the rows in this
+//       table are in the same order as the values in ShortcutIndex
+static constexpr auto allShortcutInfo = std::to_array<AllShortcutInfo>({
+    {GOTO_MEM,       ImGuiMod_Ctrl | ImGuiKey_G, true,  false, "goto_memory_address",       "Go to address in hex viewer"},
+    {STEP,           ImGuiKey_F6,                false, true,  "step",                      "Single step in debugger"},
+    {BREAK,          ImGuiKey_F7,                false, false, "break",                     "Break CPU emulation in debugger"},
+    {GOTO_DISASM,    ImGuiMod_Ctrl | ImGuiKey_G, true,  false, "goto_disasm_address",       "Go to address in dissassembler"},
+    {MOVE_UP_MEM,    ImGuiKey_UpArrow,           true,  true,  "move_up_memory_address",    "Move up in hex viewer"},
+    {MOVE_DOWN_MEM,  ImGuiKey_DownArrow,         true,  true,  "move_down_memory_address",  "Move down in hex viewer"},
+    {MOVE_LEFT_MEM,  ImGuiKey_LeftArrow,         true,  true,  "move_left_memory_address",  "Move left in hex viewer"},
+    {MOVE_RIGHT_MEM, ImGuiKey_RightArrow,        true,  true,  "move_right_memory_address", "Move right hex viewer"},
+});
+static_assert(allShortcutInfo.size() == ShortcutIndex::NUM_SHORTCUTS);
+
+std::string_view Shortcuts::getLargerDescription()
```

Comment:
I understand the intend, but it's not fully correct: we're using a proportional font, so the string with the most characters is not necessarily also the longest in pixels after rendering.

But let's keep this for now. We'll do the GUI tuning in a separate PR. (For example: we can pick a 'reasonable' default column size, make the table and window resizable, and allow scrollbars (both horizontal and vertical) to appear in the table).